src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -499,6 +494,7 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
   // This method also resets all the timer slots that have been read.
   // It is assumed to be called periodically, e.g. on `vkQueuePresentKHR`.
   void CompleteSubmits(VkDevice device) {
+    absl::WriterMutexLock lock(&mutex_);
```

Comment:
Should we really lock the complete present? I guess it is fine for now, however.

Reply:
Yes, I think the scope here can be a bit tighter, but the write lock is at least needed for the PullCompletedSubmissions call (it was there before) and the first loop here (which is potentially adding submissions back into the queue). I think I'll keep it as-is for simplicity but we can definitely keep this in mind for an improvement.

Reply:
Sounds good.